src/overlays/actors/ovl_Demo_6K/z_demo_6k.c

```diff
@@ -27,57 +53,779 @@ const ActorInit Demo_6K_InitVars = {
+            this->unk_293 = 255;
+        }
+
+        if (this->timer2 > 104) {
+            func_80967BF8(PLAYER, globalCtx);
+            Actor_Kill(&this->actor);
+            Audio_PlayActorSound2(&PLAYER->actor, NA_SE_EN_FANTOM_HIT_THUNDER);
+        } else if (this->timer2 > 94) {
+            Actor_SetScale(&this->actor, this->actor.scale.x + 0.03f);
+
+            if (this->timer2 == 95) {
+                osSyncPrintf(VT_FGCOL(CYAN) "  NA_SE_EN_GANON_FIRE_DEMO\n" VT_RST);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_FIRE_DEMO);
+            }
+        }
+
+        Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                  this->actor.world.pos.z, 255, 200, 0, this->unk_293);
+    }
+}
+
+void func_80967F10(Demo6K* this, GlobalContext* globalCtx) {
+    if (this->timer2 == 0) {
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_6K, this->actor.world.pos.x, this->actor.world.pos.y,
+                    this->actor.world.pos.z, 0, 0, 0, 13);
+    }
+
+    this->timer2++;
+
+    if (this->timer2 > 60) {
+        Actor_Kill(&this->actor);
+    }
+
+    Actor_SetScale(&this->actor, 0.05f - (this->timer2 * 0.00075f));
+}
+
+void Demo6K_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Demo6K* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void func_80967FFC(Actor* thisx, GlobalContext* globalCtx) {
+    Demo6K* this = THIS;
+    s32 pad;
+    u16 timer1 = this->timer1;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_demo_6k.c", 1070);
+
+    func_80093D84(globalCtx->state.gfxCtx);
+    Matrix_RotateX(-M_PI / 2, MTXMODE_APPLY);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0x7FFF - ((timer1 * 8) & 0x7FFF), 16, 512, 1, 0,
+                                0x7FFF - ((timer1 * 8) & 0x7FFF), 16, 32));
+
+    {
+        s32 i;
+        s32 pad;
+        Color_RGB8 colors[6][2] = {
+            { { 255, 170, 255 }, { 255, 0, 100 } }, { { 255, 255, 170 }, { 0, 255, 0 } },
+            { { 255, 255, 170 }, { 255, 255, 0 } }, { { 255, 170, 255 }, { 50, 0, 255 } },
+            { { 255, 255, 170 }, { 255, 100, 0 } }, { { 170, 255, 255 }, { 0, 100, 255 } },
+        };
+
+        Matrix_RotateZ(-M_PI / 2, MTXMODE_APPLY);
+
+        for (i = 0; i < 6; i++) {
+            Matrix_RotateZ(M_PI / 3, MTXMODE_APPLY);
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_demo_6k.c", 1115),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gDPPipeSync(POLY_XLU_DISP++);
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, colors[i][0].r, colors[i][0].g, colors[i][0].b, 255);
+            gDPSetEnvColor(POLY_XLU_DISP++, colors[i][1].r, colors[i][1].g, colors[i][1].b, 255);
+            gSPDisplayList(POLY_XLU_DISP++, D_060022B0);
+        }
+
+        // required to avoid optimizing out i
+        if ((s16)i) {}
```

Comment:
does IDO fail to optimize out a second for loop here with no body? Does that have the same effect?

Reply:
No, I can't get it to match any other way.